torch/csrc/autograd/functions/accumulate_grad.cpp

```diff
@@ -59,5 +60,45 @@ auto AccumulateGrad::apply(variable_list&& grads) -> variable_list {
   return variable_list();
 }
 
+#ifdef TORCH_COMPILED_AUTOGRAD
+void AccumulateGrad::compiled_args(CompiledNodeArgs& args) {
+  if (args.cond(
+          variable.defined() && variable.requires_grad() &&
```

Comment:
why check requires_grad here?

Reply:
This is mirroring checks in `apply()` above.  If these aren't true then the AccumulateGrad node becomes a no-op.  We need to cache on this, since it would result in a different graph.

Reply:
Ho wow, I forgot that we were checking this dynamically during backward
```python
import torch


a = torch.rand(10, requires_grad=True)
b = a.clone()

b.sum().backward()
print(a.grad) # prints a bunch of ones
a.grad=None
a.requires_grad=False
b.sum().backward()
print(a.grad) # prints None
```
cc @soulitzer interesting fact.

Reply:
Pretty cool :)